### PR TITLE
fix(cr-gateway): Remove GitHub API requirement from CR server.

### PR DESCRIPTION
**Pull request by technosophos**
Relates to #225

### CODE REVIEW

File: pkg/webhook/dockerhub.go [hunk 1]
```diff
@@ -63,34 +63,24 @@ func (s *dockerPushHook) Handle(c *gin.Context) {
...
 
-func (s *dockerPushHook) doDockerImagePush(proj *brigade.Project, commit string, payload, brigadeJS []byte) error {
+func (s *dockerPushHook) doDockerImagePush(proj *brigade.Project, commit string, payload []byte) error {
 	b := &brigade.Build{
 		ProjectID: proj.ID,
 		Type:      "image_push",
 		Provider:  "dockerhub",
 		Commit:    commit,
 		Payload:   payload,
-		Script:    brigadeJS,
```
Comment (mumoshu) [inline pkg/webhook/dockerhub.go]: Can we really just defer determining the script to be executed for a build like this?

As far as I understand, brigade-controller as of today doesn't have an ability to populate missing brigade.js(=`script` in a secret representing an brigade event) - it looks like we have to populate it on gateway-side. 

I'm writing this referring the followings:

- https://github.com/Azure/brigade/search?utf8=%E2%9C%93&q=syncSecret&type=
- https://github.com/Azure/brigade/search?utf8=%E2%9C%93&q=Script&type=

Please correct me if I'm missing something!

Comment (mumoshu) [reply pkg/webhook/dockerhub.go]: Just noticed that brigade does handle this case [here](https://github.com/Azure/brigade/blob/2d6281b12ed97d70ddb675dc79a5e70dc8722583/brigade-controller/cmd/brigade-controller/controller/handler.go#L137-L162).
Sorry for bothering!

For anyone interested, in nutshell brigade seems to create a vcs-sidecar(as an init-container) to populate `/vcs/brigade.js` only when `script` isn't included in an event. Then, the brigade-worker runs either `/etc/brigade/script`(corresponds to the `script` included in a `brigade.Build`=secret) or `/vcs/brigade.js` [according to this code](https://github.com/Azure/brigade/blob/2d6281b12ed97d70ddb675dc79a5e70dc8722583/brigade-worker/prestart.js). So, if we omit a `script` in a brigade.Build coming from the cr-gateway, the worker just tries to git-clone the necessary script=`brigade.js` from the vcs repo. No problem.

File: charts/brigade-project/values.yaml [hunk 2]
```diff
@@ -12,8 +12,7 @@ cloneURL: "https://github.com/deis/empty-testbed.git"
 sharedSecret: "IBrakeForSeaBeasts"
 
 # OPTIONAL: Use this to have Brigade update your project about the build.
-# You probably want this if you want pull requests or commits to show
-# the build status.
+# This is REQUIRED for the GitHub gateway, but optional otherwise.
```
Comment (blimmer) [inline charts/brigade-project/values.yaml]: 👍

File: pkg/webhook/dockerhub.go (hunk 1 above)
Comment (technosophos) [reply pkg/webhook/dockerhub.go]: Thanks for taking the time to look through all of that.

**Pull request merged**